torch/linalg/__init__.py

```diff
@@ -658,60 +658,69 @@
 also batched with the same batch dimensions as :attr:`input`.
 
-If :attr:`full_matrices` is ``False`` (default), the method returns the reduced singular
-value decomposition i.e., if the last two dimensions of :attr:`input` are
-``m`` and ``n``, then the returned `U` and `V` matrices will contain only
-:math:`min(n, m)` orthonormal columns.
+If :attr:`full_matrices` is ``False``, the method returns the reduced singular
+value decomposition. In this case, if the last two dimensions of :attr:`input` are
+``m`` and ``n``, then the returned ``U`` and ``Vh`` matrices will contain only
+``min(n, m)`` orthonormal columns.
 
-If :attr:`compute_uv` is ``False``, the returned `U` and `Vh` will be empy
+If :attr:`compute_uv` is ``False``, the returned ``U`` and ``Vh`` will be empty
 tensors with no elements and the same device as :attr:`input`. The
 :attr:`full_matrices` argument has no effect when :attr:`compute_uv` is False.
 
-The dtypes of ``U`` and ``V`` are the same as :attr:`input`'s. ``S`` will
+The dtypes of ``U`` and ``Vh`` are the same as :attr:`input`'s. ``S`` will
 always be real-valued, even if :attr:`input` is complex.
 
 .. note:: Unlike NumPy's ``linalg.svd``, this always returns a namedtuple of
-          three tensors, even when :attr:`compute_uv=False`. This behavior may
+          three tensors, even when :attr:`compute_uv` is ``False``. This behavior may
           change in a future PyTorch release.
 
 .. note:: The singular values are returned in descending order. If :attr:`input` is a batch of matrices,
-          then the singular values of each matrix in the batch is returned in descending order.
+          then the singular values of each matrix in the batch are returned in descending order.
 
-.. note:: The implementation of SVD on CPU uses the LAPACK routine `?gesdd` (a divide-and-conquer
-          algorithm) instead of `?gesvd` for speed. Analogously, the SVD on GPU uses the cuSOLVER routines
-          `gesvdj` and `gesvdjBatched` on CUDA 10.1.243 and later, and uses the MAGMA routine `gesdd`
-          on earlier versions of CUDA.
+.. note:: The implementation of SVD on CPU uses the LAPACK routine ``?gesdd`` (a divide-and-conquer
+          algorithm) instead of ``?gesvd`` for speed. Analogously, the SVD on GPU uses the
+          cuSOLVER routines ``gesvdj`` and ``gesvdjBatched`` on CUDA 10.1.243 and later,
+          and uses the MAGMA routine ``gesdd`` on earlier versions of CUDA.
 
-.. note:: The returned matrix `U` will be transposed, i.e. with strides
-          :code:`U.contiguous().transpose(-2, -1).stride()`.
+.. note:: The returned matrix ``U`` will not be contiguous. It will be represented as a
+          column-major matrix (i.e. fortran-contiguous).
 
-.. note:: Gradients computed using `U` and `Vh` may be unstable if
-          :attr:`input` is not full rank or has non-unique singular values.
+.. note:: Gradients computed using ``U`` and ``V`` may be unstable if :attr:`input` has
```

Comment:
What does "unstable" mean?

Reply:
Close to singular. Basically the gradient has a `1. / (s_i - s_j)` term, where `s_i` are the singular values.

Reply:
Can we clarify that in the documentation? Or maybe we should consider adding a glossary to the linalg docs?

Reply:
I've added a better explanation now as its done in the documentation of `symeig`. I will open an issue to discuss how to make the documentation of `torch.linalg` more consistent, to discuss both the style and things like this one.